Stop using CURL for http connections

calavera opened the pull request:
Hi,

I've modified the code to stop using curl in favor or Excon, a great library for long http connections. It would be great if you could merge my changes on master and release a new version of the gem with them.

In the process, I've also added the Gemfile to work with bundler and make it easy to develop.

If you don't have any further interest in continue maintaining this library, please let me know, I'll be gladly take care of it.

Cheers

calavera committed:
use excon to upload the template

calavera committed:
bundler friendly repository

calavera committed:
fix fileItem.create mistake

calavera committed:
remove curl calls once for all

calavera committed:
configure bundler for development

calavera committed:
fix double asignment

scottjg commented:
can you do the pull request on the vmware/rbvmomi repo? we don't know how to change the direction of the fork at this point, and that one is going to be considered the main repo going forward.

calavera commented:
Oh, sure, I didn't know there was another repo.

scottjg commented on lib/rbvmomi/vim/OvfManager.rb:
```diff
@@ -104,4 +95,29 @@ def deployOVF opts
+      tmp_dir = Dir.mktmpdir
+      file_body = Excon.get(tmp).body
+      file_path = File.expand_path(path, ovf_uri)
+      File.open(file_path, 'w') {|f| f.write file_body}
```
so the original behavior, using curl to download the file and pipe it to another curl to upload, did have a benefit: streaming through a pipe means that we never need to have the entire virtual disk in memory (could be many gigabytes large), or on disk.

if i'm reading this code correctly, this may use quite a bit more memory and disk space than the previous implementation (it downloads the file to disk, then reads it all into memory). excon supports streaming responses according to the readme. can you use those instead?

scottjg replied:
so the original behavior, using curl to download the file and pipe it to another curl to upload, did have a benefit: streaming through a pipe means that we never need to have the entire virtual disk in memory (could be many gigabytes large), or on disk.

if i'm reading this code correctly, this may use quite a bit more memory and disk space than the previous implementation (it downloads the file to disk, then reads it all into memory). excon supports streaming responses according to the readme. can you use those instead?
Add a line note

calavera replied:
Yeah, you're right. This code doesn't handle streaming, I can fix that. In my case it works just fine because the disk is already downloaded and I don't need that streaming. In my previous attempts using curl it didn't work streaming from a local path.

The "previous bar" was curl's output.

scottjg replied:
cool, it would be good to keep some kind of progress indication if you're going to fix the code for streaming. i liked that feature.

scottjg commented:
err.. meant to comment on the other repo. let's follow the discussion there.

calavera committed:
Do not keep the disk in local when we get it from the net. The content is
still in memory and we send it directly to vSphere. This mimics the previous
behaviour with curl.
